training/src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,95 @@
+package wolox.training.controllers;
+
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.Users;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("api/books")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+
+    @GetMapping("/users")
+    public Iterable findAll() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping("/users/{id}")
+    public Users findOne(@PathVariable Long id) {
+        return userRepository.findById(id)
+            .orElseThrow(UserNotFoundException::new);
+    }
+
+
+    @DeleteMapping("/users/{id}")
+    public void delete(@PathVariable Long id) {
+        Optional<Users> userToDelete = userRepository.findById(id);
+        if (userToDelete.isPresent()) {
+            userToDelete.get().getBooks().removeAll(userToDelete.get().getBooks());
+            userRepository.delete(userToDelete.get());
+        } else {
+            throw new UserNotFoundException();
+        }
+    }
+
+
+    @PostMapping("/user")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Users create(@RequestBody Users user) {
+        return userRepository.save(user);
+    }
+
+    @PutMapping("/users/{id}")
+    public Users updateUser(@RequestBody Users users, @PathVariable Long id) {
+        if (users.getId() != id) {
+            throw new UserNotFoundException();
+        }
+        userRepository.findById(id)
+            .orElseThrow(UserNotFoundException::new);
+        return userRepository.save(users);
+    }
+
+    @PutMapping("/users/{id_user}/book/{id_book}")
+    public Users updateBookOfUser(@PathVariable String id_user, @PathVariable String id_book) {
+        Optional<Users> user = userRepository.findById(Long.parseLong(id_user));
+        if (user.isPresent()) {
+            Optional<Book> bookToAssing = bookRepository.findById(Long.parseLong(id_book));
+            if (bookToAssing.isPresent()) {
```

Comment:
Using `orElseThrow` in these two cases would make the code cleaner and easier to read 😊 . Also, you would avoid the `Optional` so you won't need to use the `entity.get()` that is not nice at all

Reply:
but in that case i think that i need the variable user and bookToAssing for check if the book is in the list of books of user or am I wrong?

Reply:
Yes, you need them but you could do this:
`Users user = userRepository.findById(Long.parseLong(id_user)).orElseThrow(UserNotFound::new);`
and then use the `user` variable. You know it won't be empty because if the user was not found, an exception would have been thrown. 
Same with the `book` variable.